src/System.Text.Json/src/System/Text/Json/Writer/Utf8JsonWriter.cs

```diff
@@ -62,7 +62,10 @@ public sealed partial class Utf8JsonWriter : IDisposable, IAsyncDisposable
         /// Returns the amount of bytes written by the <see cref="Utf8JsonWriter"/> so far
         /// that have not yet been flushed to the output and committed.
         /// </summary>
-        public int BytesPending { get; private set; }
+        public int BytesPending => _bytesPending + _currentIndex;
```

Comment:
Is there a cleaner way to achieve this without introducing an extra field, or is this fine?

`BytesPending` generally isn't called a lot. It is used in the serializer loop but that is doing a lot more work (so an extra addition shouldn't be too problematic - I avoided branches).
https://github.com/dotnet/corefx/blob/1841042b99062de13dc80098cede9413be569238/src/System.Text.Json/src/System/Text/Json/Serialization/JsonSerializer.Write.cs#L67-L71

cc @ steveharter